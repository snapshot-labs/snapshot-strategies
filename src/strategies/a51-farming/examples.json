[
  {
    "name": "Example query",
    "strategy": {
      "name": "a51-farming",
      "params": {
        "farmingVaultAddress": "0x5ea84bb07f399fbf58c504929864b5fcbdc8300b",
        "isToken0": false,
        "decimals": 18
      }
    },
    "network": "137",
    "addresses": [
      "0x0895fcb879d7b0d36f23b3f68b6192c532ffad01",
      "0x072fb32a9d71482d970090daef0e90cd8c665113",
      "0x75f2BbF3A10B67134A780fB7D6da5808265111F7"
    ],
    "snapshot": 55932446
  }
]
