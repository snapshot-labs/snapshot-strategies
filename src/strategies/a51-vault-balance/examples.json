[
  {
    "name": "Example query",
    "strategy": {
      "name": "a51-vault-balance",
      "params": {
        "vaultAddress": "0x2ba1bc7d00ed1cb31379ee28a88968d90a6c2f49",
        "isToken0": false,
        "decimals": 18
      }
    },
    "network": "137",
    "addresses": [
      "0x22b77d7135174e1002d543ef8ff16a775db9ae5f",
      "0x75f2bbf3a10b67134a780fb7d6da5808265111f7",
      "0xfcae8fbaabb5336d8f4ddd76e6542fd6b89be68e"
    ],
    "snapshot": 55932446
  }
]
