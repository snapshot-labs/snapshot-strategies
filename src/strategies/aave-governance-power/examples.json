[
  {
    "name": "Example of Aave Governance Power strategy",
    "strategy": {
      "name": "aave-governance-power",
      "params": {
        "symbol": "AAVE+stkAAVE",
        "decimals": 18,
        "powerType": "vote",
        "governanceStrategy": "0xb7e383ef9b1e9189fc0f71fb30af8aa14377429e"
      }
    },
    "network": "1",
    "addresses": [
      "0x329c54289Ff5D6B7b7daE13592C6B1EDA1543eD4",
      "0x57ab7ee15cE5ECacB1aB84EE42D5A9d0d8112922",
      "0x0ab97008cad303a8C90ea630c282760284c19e93"
    ],
    "snapshot": 18054498
  }
]
