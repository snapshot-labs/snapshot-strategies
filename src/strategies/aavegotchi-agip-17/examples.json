[
  {
    "name": "AGIP 17: Voting power for GHST value of parcels",
    "strategy": {
      "name": "aavegotchi-agip-17",
      "params": {
        "symbol": "REALM"
      }
    },
    "network": "137",
    "addresses": ["0x027Ffd3c119567e85998f4E6B9c3d83D5702660c"],
    "snapshot": 22089223
  }
]
