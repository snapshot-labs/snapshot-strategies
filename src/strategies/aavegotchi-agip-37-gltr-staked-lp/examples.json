[
  {
    "name": "AGIP 37: Voting Power for GHST-FUD, GHST-FOMO, GHST-ALPHA, GHST-KEK, GHST-GLTR, GHST-USDC, and GHST-WMATIC staked in the GLTR farming contract. Note: Excludes wapGHST.",
    "strategy": {
      "name": "aavegotchi-agip-37-gltr-staked-lp",
      "params": {
        "ghstAddress": "0x385Eeac5cB85A38A9a07A70c73e0a3271CfB54A7",
        "gltrStakingAddress": "0x1fE64677Ab1397e20A1211AFae2758570fEa1B8c",
        "amGhstAddress": "0x080b5BF8f360F624628E0fb961F4e67c9e3c7CF1",
        "wapGhstAddress": "0x73958d46B7aA2bc94926d8a215Fa560A5CdCA3eA",
        "wapGhstPoolId": 0,
        "ghstFudAddress": "0xfec232cc6f0f3aeb2f81b2787a9bc9f6fc72ea5c",
        "ghstFudPoolId": 1,
        "ghstFomoAddress": "0x641ca8d96b01db1e14a5fba16bc1e5e508a45f2b",
        "ghstFomoPoolId": 2,
        "ghstAlphaAddress": "0xc765eca0ad3fd27779d36d18e32552bd7e26fd7b",
        "ghstAlphaPoolId": 3,
        "ghstKekAddress": "0xbfad162775ebfb9988db3f24ef28ca6bc2fb92f0",
        "ghstKekPoolId": 4,
        "ghstUsdcAddress": "0x096c5ccb33cfc5732bcd1f3195c13dbefc4c82f4",
        "ghstUsdcPoolId": 5,
        "ghstWmaticAddress": "0xf69e93771F11AECd8E554aA165C3Fe7fd811530c",
        "ghstWmaticPoolId": 6,
        "ghstGltrAddress": "0xb0E35478a389dD20050D66a67FB761678af99678",
        "ghstGltrPoolId": 7,
        "symbol": "GHST",
        "decimals": 18
      }
    },
    "network": "137",
    "addresses": [
      "0x26cf02F892B04aF4Cf350539CE2C77FCF79Ec172",
      "0x027Ffd3c119567e85998f4E6B9c3d83D5702660c",
      "0xc4cb6cb969e8b4e309ab98e4da51b77887afad96"
    ],
    "snapshot": 33423178
  }
]
