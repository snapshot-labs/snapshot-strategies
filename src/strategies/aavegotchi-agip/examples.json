[
  {
    "name": "AGIP 8+9: Voting power for GHST value of items, their aavegotchis equippedWearables + baseRarityScore",
    "strategy": {
      "name": "aavegotchi-agip",
      "params": {
        "tokenAddress": "0x86935f11c86623dec8a25696e1c19a8659cbf95d",
        "symbol": "GOTCHI"
      }
    },
    "network": "137",
    "addresses": [
      "0x51195e21BDaE8722B29919db56d95Ef51FaecA6C",
      "0xDd564df884Fd4e217c9ee6F65B4BA6e5641eAC63",
      "0xBfe09443556773958bae1699b786d8E9680B5571"
    ],
    "snapshot": 39986904
  }
]
