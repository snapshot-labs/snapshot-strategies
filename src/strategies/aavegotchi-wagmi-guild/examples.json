[
  {
    "name": "Voting power based on aavegotchis equipped with wagie wearables for the WAGMI Warriors Guild",
    "strategy": {
      "name": "aavegotchi-wagmi-guild",
      "params": {
        "tokenAddress": "0x86935f11c86623dec8a25696e1c19a8659cbf95d"
      }
    },
    "network": "137",
    "addresses": ["0x26cf02F892B04aF4Cf350539CE2C77FCF79Ec172"],
    "snapshot": 20122217
  }
]
