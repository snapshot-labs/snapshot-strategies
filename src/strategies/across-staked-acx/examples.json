[
  {
    "name": "Example of Across Staked-ACX Strategy",
    "strategy": {
      "name": "across-staked-acx",
      "params": {
        "acceleratingDistributorAddress": "0x9040e41eF5E8b281535a96D9a48aCb8cfaBD9a48",
        "acxLpTokenAddress": "0xb0C8fEf534223B891D4A430e49537143829c4817",
        "wethLpTokenAddress": "0x28F77208728B0A45cAb24c4868334581Fe86F95B",
        "usdcLpTokenAddress": "0xC9b09405959f63F72725828b5d449488b02be1cA",
        "wbtcLpTokenAddress": "0x59C1427c658E97a7d568541DaC780b2E5c8affb4",
        "daiLpTokenAddress": "0x4fabacac8c41466117d6a38f46d08ddd4948a0cb",
        "hubPoolAddress": "0xc186fa914353c44b2e33ebe05f21846f1048beda",
        "acxTokenAddress": "0x44108f0223A3C3028F5Fe7AEC7f9bb2E66beF82F"
      }
    },
    "network": "1",
    "addresses": [
      "0x9A8f92a830A5cB89a3816e3D267CB7791c16b04D",
      "0x718648C8c531F91b528A7757dD2bE813c3940608",
      "0x996267d7d1B7f5046543feDe2c2Db473Ed4f65e9",
      "0xbfb496ACb99299e9eCE84B3FD1B3fDd0f6CDDf49"
    ],
    "snapshot": 16171362
  }
]
