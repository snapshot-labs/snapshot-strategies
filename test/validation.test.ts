import snapshot from '../src';
import examples from '../src/validations/passport/examples.json';

const [example] = examples;
const id = 'passport';

describe('validation', () => {
  it(`validate: ${id} "${example.name}"`, async () => {
    const validation = new snapshot.validations[id](
      example.author,
      example.space,
      example.network,
      'latest',
      example.params
    );
    expect(await validation.validate()).toBe(true);
  }, 10e3);
});
