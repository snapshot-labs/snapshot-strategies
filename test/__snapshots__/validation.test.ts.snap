// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` validation 1`] = `true`;
